### ISSUE TITLE
Ouroboros is Unsound -- RUSTSEC-2023-0042

### ISSUE THREAD

**Issue by wookietreiber** [jonhoo/rust-imap#264]
Unfortunate news, but may be relevant before v3 release: https://rustsec.org/advisories/RUSTSEC-2023-0042

```console
$ cargo audit
...
Crate:     ouroboros
Version:   0.15.6
Warning:   unsound
Title:     Ouroboros is Unsound
Date:      2023-06-11
ID:        RUSTSEC-2023-0042
URL:       https://rustsec.org/advisories/RUSTSEC-2023-0042
Dependency tree:
ouroboros 0.15.6
└── imap 3.0.0-alpha.10
...
```

**Comment by wookietreiber** [comment]
Might be fixed by https://github.com/jonhoo/rust-imap/pull/265.

**Comment by jonhoo** [comment]
#265 is now merged and according to the advisory, 0.16 does not have this issue any more, so I think we're good! I'll cut a new alpha soon-ish :) If I forget, please ping me!
